test/manual/kerberos.test.js

```diff
@@ -51,6 +51,16 @@ describe('Kerberos', function () {
     });
   });
 
+  it('validate that CANONICALIZE_HOST_NAME can be passed in', function (done) {
+    const client = new MongoClient(
+      `${krb5Uri}&authMechanismProperties=SERVICE_NAME:mongodb,CANONICALIZE_HOST_NAME:true&maxPoolSize=1`
+    );
+    client.connect(function (err, client) {
+      if (err) return done(err);
+      verifyKerberosAuthentication(client, done);
+    });
```

Comment:
`verifyKerberosAuthentication` just asserts that we can connect and run an operation, is there more we could do to check that canonicalization was performed? sinon should be able to listen to `dns.resolveCname` maybe?

Reply:
I've added a spy in these tests now to verify the method was called.